driver/src/orderbook/streamed/error.rs

```diff
@@ -8,6 +9,12 @@ pub enum Error {
     UnknownOrder,
     #[error("order already exists")]
     OrderAlreadyExists,
-    #[error("math underflow")]
-    MathUnderflow,
+    #[error("math under or overflow")]
+    MathOverflow,
+    #[error("withdraw batch does not match withdraw request {0}")]
+    WithdrawEarlierThanRequested(BatchId),
+    #[error("withdraw amount is larger than withdraw request {0}")]
+    WithdrawMoreThanRequested(U256),
+    #[error("withdraw amount is larger than balance {0}")]
+    WithdrawMoreThanBalance(U256),
```

Comment:
I think it might be worth wrapping this error type in something that identifies the event (by block number and log index for example) so that the exact event that caused the problem can be inspected on, say, etherscan.

Reply:
For a separate PR though!

Reply:
As I said above, this should be done by the "state manager" because it knows on which event the error happened and can add that information.

Reply:
Perfect! That was exactly what I meant.